mm2src/coins/utxo/rpc_clients.rs

```diff
@@ -2320,12 +2320,25 @@ impl UtxoRpcClientOps for ElectrumClient {
     }
 
     fn display_balance(&self, address: Address, decimals: u8) -> RpcRes<BigDecimal> {
-        let hash = electrum_script_hash(&output_script(&address, ScriptType::P2PKH));
-        let hash_str = hex::encode(hash);
-        Box::new(
-            self.scripthash_get_balance(&hash_str)
-                .map(move |electrum_balance| electrum_balance.to_big_decimal(decimals)),
-        )
+        let this = self.clone();
+        let mut hashes = Vec::new();
+        // If the plain pubkey is available, fetch the balance found in P2PK as well (if any).
+        if address.hash.is_pubkey() {
+            hashes.push(hex::encode(electrum_script_hash(&output_script(&address, ScriptType::P2PK))));
+        }
+        hashes.push(hex::encode(electrum_script_hash(&output_script(&address, ScriptType::P2PKH))));
```

Comment:
As discussed in DM, this will show double the balance for segwit addresses. Because of this
```rust
pub fn output_script(address: &Address, script_type: ScriptType) -> Script {
    match address.addr_format {
        UtxoAddressFormat::Segwit => Builder::build_p2witness(&address.hash),
```
Refactoring of the output script in the other PR should fix this issue.